File: frontend/src/types/Location.ts
export interface LocationDTO {
  address: string;
  latitude: number;
  longitude: number;
}

export const DEFAULT_LOCATION: LocationDTO = {
  address: "",
  latitude: -34.6037,
  longitude: -58.3816,
};

export interface BusinessAccountDTO {
  id: string;
  name: string;
  description?: string;
  location: LocationDTO;
}

File: backend/src/main/java/com/tripmates/backend/common/types/Location.java
package com.tripmates.backend.common.types;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.swagger.v3.oas.annotations.media.Schema;

@JsonIgnoreProperties(ignoreUnknown = true)
public record Location(@Schema(description = "Location's physical address") @JsonProperty("address") String address,
		@Schema(description = "Location's latitude coordinate") @JsonProperty("latitude") Double latitude,
		@Schema(description = "Location's longitude coordinate") @JsonProperty("longitude") Double longitude) {

	@JsonCreator
	public Location(@JsonProperty("address") String address, @JsonProperty("latitude") Double latitude,
			@JsonProperty("longitude") Double longitude) {
		this.address = address;
		this.latitude = latitude;
		this.longitude = longitude;
	}

}